effekt/shared/src/main/scala/effekt/core/StaticArguments.scala

```diff
@@ -203,20 +204,20 @@ object StaticArguments {
     val statics: Map[Id, IsStatic] = recursiveFunctions.map {
       case (id, RecursiveFunction(BlockLit(tparams, cparams, vparams, bparams, body), targs, vargs, bargs)) =>
         val isTypeStatic = tparams.zipWithIndex.collect {
-          case (param, index) => targs.map(args => args(index)).forall {
+          case (param, index) => targs.nonEmpty && targs.map(args => args(index)).forall {
```

Comment:
We transformed also functions that do not have any recursive calls, so I now explicitly check whether there are arguments at all.